### PR TITLE
feat: Add UMA foundry fixtures

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

Before we start writing more tests in foundry we need some fixtures and setup points. This PR adds the required UMA fixtures for the Optimistic asserter.

### CODE REVIEW

File: packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol [hunk 1]
```diff
@@ -26,48 +27,69 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface {
 
     bytes32 identifier = "ASSERT_TRUTH";
 
-    constructor(address _finderAddress) {
-        finder = FinderInterface(_finderAddress);
+    IERC20 public defaultCurrency;
```
Comment (chrismaree) [inline packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:30]: I did: a) moved this file to a different directory b) add default props c) added a function to set these defaults.

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Ownership.t.sol [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (chrismaree) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Ownership.t.sol:1]: this is the key starting point that you guys should look at: this is how simple it should be to setup a test and run the base tests. I've re-structured things to pass around structs of contracts from each set of fixtures. see below.

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Ownership.t.sol [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+pragma solidity ^0.8.0;
+
+import "../fixtures/optimistic-assertor/SetUpOptimisticAssertor.sol";
+
+contract ContractTest is Test {
+    SetUpOptimisticAssertor.OptimisticAsserterContracts umaContracts;
+    OptimisticAssertor optimisticAssertor;
```
Comment (chrismaree) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Ownership.t.sol:8]: note we could probs ditch this second state variable to be even cleaner but I just wanted to show how to access the other contracts from the fixture.

**Review by md0x — APPROVED** [review]
Lgtm

**Review by Reinis-FRP — APPROVED** [review]
Great! Just fixed case sensitivity on imports to be compatible with linux filesystems

**Pull request merged**